android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothQNScale.java

```diff
@@ -87,12 +95,29 @@ public String driverName() {
     protected boolean onNextStep(int stepNr) {
         switch (stepNr) {
             case 0:
+                // Try writing bytes to 0xffe4 to check whether to use 1st or 2nd type
+                try {
+                    long timestamp = new Date().getTime() / 1000;
+                    timestamp -= SCALE_UNIX_TIMESTAMP_OFFSET;
+                    byte[] date = new byte[4];
+                    Converters.toInt32Le(date, 0, timestamp);
+                    writeBytes(WEIGHT_MEASUREMENT_SERVICE, CUSTOM4_MEASUREMENT_CHARACTERISTIC, new byte[]{(byte) 0x02, date[0], date[1], date[2], date[3]});
+                } catch (NullPointerException e) {
+                    useFirstType = false;
+                }
+
                 // set notification on for custom characteristic 1 (weight, time, and others)
-                setNotificationOn(WEIGHT_MEASUREMENT_SERVICE, CUSTOM1_MEASUREMENT_CHARACTERISTIC);
```

Comment:
maybe we have here a timing problem as the above test with the catch ... statement will take some time!? Maybe somebody could try to add an extra step to setup the notifcations on?